pkg/resources/scaleway/resources.go

```diff
@@ -0,0 +1,106 @@
+package scaleway
+
+import (
+	"k8s.io/kops/pkg/resources"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/scaleway"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+)
+
+const (
+	resourceTypeDNSRecord    = "dns-record"
+	resourceTypeGateway      = "gateway"
+	resourceTypeLoadBalancer = "load-balancer"
+	resourceTypeVolume       = "volume"
+	resourceTypeServer       = "server"
+	resourceTypeVPC          = "vpc"
+)
+
+type listFn func(fi.Cloud, string) ([]*resources.Resource, error)
+
+func ListResources(cloud scaleway.ScwCloud, clusterInfo resources.ClusterInfo) (map[string]*resources.Resource, error) {
+	resourceTrackers := make(map[string]*resources.Resource)
+	clusterName := clusterInfo.Name
+
+	listFunctions := []listFn{
+		listServers,
+		listVolumes,
```

Comment:
SSH keys?

Reply:
I trimmed the PR to keep it shorter, I was going to just add the function to list and delete ssh keys but by doing that I noticed that our account API that I was using will soon be deprecated and is being replaced by our IAM API. I should probably make an other PR to do the migration right away before adding the ListSSHKeys function to this one.

Reply:
OK